src/tools/cephfs/cephfs-shell

```diff
@@ -306,16 +323,17 @@ class CephFSShell(Cmd):
         """    
         return self.complete_filenames(text, line, begidx, endidx)
 
-    mkdir_parser = argparse.ArgumentParser(description = 'Create the directory(ies), if they do not already exist.')
-    mkdir_parser.add_argument('dirs', type = str, metavar = 'DIR_NAME', help = 'Name of new_directory.', nargs = '+')
-    mkdir_parser.add_argument('-m', '--mode', action = 'store', help = 'Sets the access mode for the new directory.', type = str)
-    mkdir_parser.add_argument('-p', '--parent', action = 'store_true', help = 'Create parent directories as necessary. When this option is specified, no error is reported if a directory already exists.')
-    
-    @with_argparser(mkdir_parser)
     def do_mkdir(self, args):
         """
         Create directory.
-        """    
+        """
+        global mkdir_parser 
+        if mkdir_parser is None:
+            mkdir_parser = argparse.ArgumentParser(description = 'Create the directory(ies), if they do not already exist.')
+            mkdir_parser.add_argument('-m', '--mode', action = 'store', help = 'Sets the access mode for the new directory.', type = str)
+            mkdir_parser.add_argument('-p', '--parent', action = 'store_true', help = 'Create parent directories as necessary. When this option is specified, no error is reported if a directory already exists.')
+            mkdir_parser.add_argument('dirs', type = str, help = 'Name of new_directory.', nargs = '+')
+        args = mkdir_parser.parse_args(shlex.split(args))
```

Comment:
@rpavani1998 i think we can also have a re-implementation of `with_argparser` . also, i'd suggest put the `description` of a argparser as the `do_` method's doc string.

```py
try:
    from cmd2 import with_argparser
except ImportError:
    def with_argparser(argparser):
        import functools
        def argparser_decorator(func):
            @functools.wraps(func)
            def wrapper(thiz, cmdline):
                # do not split if it's already a list
                if not isinstance(cmdline, list):
                    arglist = shlex.split(cmdline, posix=False)
                    # in case user quotes the command args
                    arglist = [arg.strip('\'""') for arg in arglist]
                try:
                    args = argparser.parse_args(arglist)
                except SystemExit:
                    # argparse exits at seeing bad arguments
                    return
                else:
                    return func(thiz, args)

            argparser.prog = func.__name__[3:]
            if argparser.description is None and func.__doc__:
                argparser.description = func.__doc__

            return wrapper

        return argparser_decorator
```

Reply:
Yaa..That's a better approach. Thanks. I will re-implement with_argparser.